docs/user-guide/101-library.md

```diff
@@ -0,0 +1,109 @@
+---
+title: "Library"
+tag: "user-guide"
+render_with_liquid: false
+---
+
+The NMPolicy project can be integrated as a golang library calling the public 
+function [GenerateState](https://pkg.go.dev/github.com/nmstate/nmpolicy/nmpolicy#GenerateState).
+
```

Comment:
Perhaps point to the policy syntax, so someone that reads the below will have the opertunity to read the meaning of each section/field.

Reply:
Well a lib integrator need the golang docs about the structs, maybe we can reference the syntax at the golang docs ?

Reply:
I am not sure I understand. I just pointed out that you give here an example, but if the reader reaches this place before the policy syntax page, he will understand nothing from the snippet. You could just add a note to see the syntax detail at ...

Reply:
Ahh I misunderstood, adding a ref to the syntax.